src/DataFrame/IO/Parquet/Page.hs

```diff
@@ -94,6 +94,8 @@ readPageHeader hdr xs lastFieldId =
 
 readPageTypeHeader :: PageTypeHeader -> [Word8] -> Int16 -> (PageTypeHeader, [Word8])
 readPageTypeHeader hdr [] _ = (hdr, [])
+readPageTypeHeader INDEX_PAGE_HEADER _ _  = undefined
```

Comment:
If we ever do encounter an index page header I think we want to skip the strict bu we can do in a follow up. Please add a todo